test/integration/disruption/disruption_test.go

```diff
@@ -508,13 +514,14 @@ func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podN
 }
 
 func TestPatchCompatibility(t *testing.T) {
-	s, pdbc, _, clientSet, _, _ := setup(t)
-	defer s.TearDownFn()
-
 	// Even though pdbc isn't used in this test, its creation is already
 	// spawning some goroutines. So we need to run it to ensure they won't leak.
-	ctx, cancel := context.WithCancel(context.Background())
-	cancel()
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	s, pdbc, _, clientSet, _, _ := setup(ctx, t)
+	defer s.TearDownFn()
+	defer cancel()
```

Comment:
This is not the same as before: previously, cancel was called immediately. Apparently that is what the comment is trying to explain (not entirely obvious to me, though).

Reply:
We might need two cancel contexts here: one for `pdbc.Run` which is already canceled when that function gets called, then another that gets canceled after the test for the other calls.

Reply:
> This is not the same as before: previously, cancel was called immediately. Apparently that is what the comment is trying to explain (not entirely obvious to me, though).

Since pdbc is not used in this test, instead of using  ctx `_, ctx := ktesting.NewTestContext(t)`, I revert to the previous code `ctx, cancel := context.WithCancel(context.Background())`.

Reply:
> We might need two cancel contexts here: one for `pdbc.Run` which is already canceled when that function gets called, then another that gets canceled after the test for the other calls.

This way is also OK.

> then another that gets canceled after the test for the other calls.

another cancel context function can be written directly into the `t.Run` function and canceled after a single test run.

This can be done later, in this PR I tend to minimize the code changes first(see https://github.com/kubernetes/kubernetes/pull/119147#discussion_r1260526874). WDYT？@pohly

Reply:
Makes sense.